plugin/storage/es/mappings/mapping.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mappings
+
+import (
+	"bytes"
+
+	"github.com/jaegertracing/jaeger/pkg/es"
+)
+
+// MappingBuilder holds parameters required to render an elasticsearch index template
+type MappingBuilder struct {
+	templateBuilder es.TemplateBuilder
+	shards          int64
+	replicas        int64
+	esVersion       uint
+	esPrefix        string
+	useILM          bool
+}
+
+// NewBuilder constructs and returns an initialized MappingBuilder
+func NewBuilder(tb es.TemplateBuilder, shards, replicas int64, esVersion uint, esPrefix string, useILM bool) *MappingBuilder {
```

Comment:
I suggest simply declaring all the fields public so that the caller can initialize them directly. This method does not do anything useful, but has a hard to decipher signature at the call sites

Reply:
Agreed. I thought of initializing mapping as well in the constructor - dint pan out. I ll just make fields public as suggested.

Reply:
Done